### PR TITLE
Add selected_at to subjects

### PR DESCRIPTION
**Pull request by eatyourgreens**
Add selected_at to the subject model, and copy it across to new classifications.

Package:
lib-classifier

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
@camallen I wasn't sure what type the new attribute is, so I've guessed at Date.

**Comment by camallen** [comment]
> @camallen I wasn't sure what type the new attribute is, so I've guessed at Date.

It is a DateTime string in UTC `selected_at: "2019-04-11T11:16:29.063Z"`

**Comment by eatyourgreens** [comment]
Hopefully that's covered by `types.Date`. If this doesn't work, we can change the model to use a string.

**Comment by shaunanoordin** [comment]
> Hopefully that's covered by types.Date. If this doesn't work, we can change the model to use a string.

Would it help if we updated the tests, e.g. so `Classification.spec.js` checks if `"2019-04-11T11:16:29.063Z"` doesn't choke up `types.Date`? This might also be useful for other devs who'd otherwise need to guess what the input would look like.

**Comment by srallen** [comment]
I think `types.Date` will fail with the UTC strings. It's expecting `Date` instance (i.e. `new Date()`)

**Comment by srallen** [comment]
Which is a bit annoying. We could define our own custom types to use here as well as the tutorial seen timestamp...

**Comment by eatyourgreens** [comment]
@shaunanoordin that's a good idea about mocking up a Panoptes subject in the tests and passing it to a new Subject model.

**Comment by eatyourgreens** [comment]
Trying out Shaun's suggestion broke the model, so I'll change the type.
```
Error: [mobx-state-tree] Error while converting `{"id":"3","already_seen":false,"favorite":false,"finished_workflow":false,"locations":[{"image/png":"https://foo.bar/kctnr.png"},{"image/png":"https://foo.bar/nzvao.png"}],"metadata":{},"retired":false,"selected_at":"2019-04-11T11:16:29.063Z","selection_state":"","user_h......:false}` to `SubjectResource`:

    at path "/selected_at" value `"2019-04-11T11:16:29.063Z"` is not assignable to type: `(Date | undefined)` (No type is applicable for the union).
    at path "/selected_at" value `"2019-04-11T11:16:29.063Z"` is not assignable to type: `(Date | undefined)` (Value is not a Date or a unix milliseconds timestamp).
    at path "/selected_at" value `"2019-04-11T11:16:29.063Z"` is not assignable to type: `(Date | undefined)` (Value is not a undefined).
```

**Comment by eatyourgreens** [comment]
I've updated this to add `selected_at` to the tests too.

File: packages/lib-classifier/src/store/Classification.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ const ClassificationMetadata = types.model('ClassificationMetadata', {
     already_seen: types.optional(types.boolean, false),
     finished_workflow: types.optional(types.boolean, false),
     retired: types.optional(types.boolean, false),
+    selected_at: types.maybe(types.Date),
```
Comment (srallen) [inline packages/lib-classifier/src/store/Classification.js]: I think this will need to change to `types.maybe(types.string))` too?

Comment (eatyourgreens) [reply packages/lib-classifier/src/store/Classification.js]: Should be fixed now.

Comment (eatyourgreens) [reply packages/lib-classifier/src/store/Classification.js]: Should be fixed now.

**Pull request merged**